### PR TITLE
Refector application collection endpoints to not return full Instance objects

### PR DESCRIPTION
**Pull request by knolleary**
## Description

A pair of end points that return lists of Applications and their instances was including the *full* instance object. This is was far more information than was needed.

This PR changes those endpoints to run an `InstanceSummary` view which is the minimum information needed for list views.

Affected endpoints:
 - `/api/v1/application/:id/instances`
 - `/api/v1/teams/:teamId/applications`

I have verified all of the users of these endpoints in the UI work with these summaries.

Work to document these changes will happen as part of the api doc work in #1556 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by Pezmc** [comment]
I was worried this would cause troubles in the UI with the cached objects not being full objects any more (with regards to startus, last deploy, etc...), but the work to add an explicit status endpoint seems to have prevented that from being a problem.

File: forge/db/views/Project.js [hunk 1]
```diff
@@ -91,14 +91,38 @@ module.exports = {
             return result
         }))
     },
+    instancesSummaryList: function (app, instancesArray) {
+        return instancesArray.map((instance) => {
+            // Full settings are not
```
Comment (Pezmc) [inline forge/db/views/Project.js:96]: [nb] Not sure what this comment is trying to say!

**Review by Pezmc — APPROVED** [review]

File: forge/db/views/Project.js (hunk 1 above)
Comment (knolleary) [inline forge/db/views/Project.js:96]:
```suggestion
```

Comment (knolleary) [reply forge/db/views/Project.js]: Comment removed

**Pull request merged**